tests/nonmem/test_nonmem_model.py

```diff
@@ -1007,6 +1008,27 @@ def test_convert_model(testdata):
     assert model.model_code == correct
 
 
+def test_table_long_ids(testdata):
+    code = f"""$PROBLEM base model
+    $INPUT ID DV TIME
+    $DATA {testdata / "nonmem" / "pheno.dta"} IGNORE=@
+
+    $PRED
+    Y = THETA(1) + ETA(1) + EPS(1)
+
+    $THETA 1  ; TH1
+    $OMEGA 2 ; OM1
+    $SIGMA 3 ; SI1
+    $ESTIMATION METHOD=1 INTER
+    """
+    model = Model.create_model(StringIO(code))
+    dataset_new = model.dataset.copy()
+    dataset_new['ID'] = dataset_new['ID'] * 10000
+    model.dataset = dataset_new
+    set_estimation_step(model, 'FO', residuals=['CWRES'])
+    assert 'FORMAT=' in model.model_code
```

Comment:
Would it make sense to also check that adding `FORMAT` actually solves the issue?

Reply:
I think the only way to test if it solves the issue is by running this in NONMEM, reading the output table and making sure we have an expected numbers of IDs, I guess we could add an integration test but maybe it's enough to see that Pharmpy adds it when appropriate? Not sure